Table.h

```diff
@@ -0,0 +1,87 @@
+#ifndef HEADER_Table
+#define HEADER_Table
+/*
+htop - Table.h
+(C) 2004,2005 Hisham H. Muhammad
+(C) 2023 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h" // IWYU pragma: keep
+
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/time.h>
+#include <sys/types.h>
+
+#include "Hashtable.h"
+#include "Object.h"
+#include "RichString.h"
+#include "Vector.h"
+
+
+struct Row_;
+struct Panel_;
+struct Machine_;
+struct Table_;
+
+typedef void (*Table_scan)(struct Table_ *);
+typedef void (*Table_foreachRowFn)(struct Table_*, struct Row_*, int, void*);
+
+typedef struct Table_ {
+   Vector* rows;          /* all known; sort order can vary and differ from display order */
+   Vector* displayList;   /* row tree flattened in display order (borrowed); updated
+                             in Table_updateDisplayList when rebuilding panel */
+   Hashtable* table;      /* fast known row lookup by identifier */
+
+   struct Machine_* host;
+   const char* incFilter;
+   bool needsSort;
+   int following;         /* -1 or row being visually tracked in the user interface */
+
+   Table_scan scan;
```

Comment:
Should be virtual …

Reply:
Yep, I'm totally reworking this aspect - this function will be removed/reworked to be more in the htop OO style (we need a couple of other Table methods too, I'm finding).

Reply:
This is resolve now - Table became a "proper" base class with its own additional methods relating to Row scanning, and all the remaining presentation-related code moved from ProcessList into Table as a result.  Looks much better now IMO, thanks @BenBE